src/table/utils/styling-utils.js

```diff
@@ -110,6 +110,33 @@ export function getBodyStyle(layout, theme) {
   };
 }
 
+/**
+ * We can set the background color expression or text color expression\
+ * for measure data or dimension data.
+ * Ex:
+ * {"qHyperCubeDef": {
+ *     "qDimensions": [{
+ *        "qAttributeExpressions": [
+          {
+            "qExpression": "rgb(4,4,4)",
+            "qAttribute": true,
+            "id": "cellBackgroundColor"
+          },
+          {
+            "qExpression": "rgb(219, 42, 42)",
+            "qAttribute": true,
+            "id": "cellForegroundColor"
+          }
+        ],
+ *     }]
+ * }}
+ *
+ * get style from qAttributeExpressions in qDimensions or qMeasures
+ * @param {Object} styling - style from styling in CellRenderer in TableBodyWrapper
+ * @param {?Object} qAttrExps - qAttributeExpressions from each cell
```

Comment:
should the ? be here?

Reply:
if you want to indicate that it is optional, it should be `Object=`

Reply:
it is not optional, I want to indicate that it can be null
<img width="1190" alt="Screenshot 2022-02-25 at 16 36 31" src="https://user-images.githubusercontent.com/4471489/155743310-4dc0921e-0dbd-4373-8b31-4ece03526a11.png">
l

Reply:
ah cool good to know